Core/Generators/CPlusPlus/CPlusPlusGenerator.cs

```diff
@@ -271,6 +271,7 @@ private string CompileDecodeField(TypeBase type, string target, int depth = 0, i
                 },
                 DefinedType dt when Schema.Definitions[dt.Name] is EnumDefinition =>
                     $"{target} = static_cast<{dt.Name}>(reader.readUint32());",
+                DefinedType dt when isOptional => $"{target}.emplace({dt.Name}::decode(reader));",
```

Comment:
The fix is this line @apecoraro. Calling [`emplace`](https://en.cppreference.com/w/cpp/utility/optional/emplace) on the optional value achieves a similar thing to initializing it and then writing into `.value()` but it is somewhat more efficient, at least as far as I understand 🙂 .

Reply:
So the generated code now looks like 

```cpp
        case 1:
          target.params.emplace(AvailableCodecRequestParams::decode(reader));
          break;
```

Reply:
Nice!